server/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -93,17 +106,16 @@ public boolean implies(ProtectionDomain domain, Permission permission) {
             }
         }
 
-        if (permission instanceof FilePermission) {
```

Comment:
IIRC this was here to avoid the (potentially relatively expensive) implies call on a permission collection. Can we add this conditional back, and also guard the new check in the same way?